lms/djangoapps/courseware/access.py

```diff
@@ -469,6 +469,10 @@ def _has_group_access(descriptor, user, course_key):
         # via updating the children of the split_test module.
         return ACCESS_GRANTED
 
+    # Allow staff group access, unless they are masquerading as a student.
+    if get_user_role(user, course_key) != 'student':
```

Comment:
What are the potential user roles? It seems like this should be a positive test for staff, rather than just that the user is not a student. Do all non-student roles need to be given full access?

Also, at some point this code should be upgraded to use the best practice of not checking for roles (not now, I imagine):

http://open-edx-proposals.readthedocs.io/en/latest/oep-0009.html